app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -586,6 +592,18 @@ open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus: Bus,
     routeManager.fetchRoute(callback)
   }
 
+  override fun onRouteSuccess(route: Route) {
```

Comment:
`MainPresenterImpl` also implements `RouteCallback` itself since reroute logic is handled in this class.

https://github.com/mapzen/eraser-map/blob/master/app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt#L467

Seems like there may be an opportunity to combine this method and `success(Route)` or maybe at least have one invoke the other. This could possibly avoid some code dupe and/or divergent behavior for route vs. reroute?